core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -290,12 +290,12 @@ object AdminUtils extends Logging {
     var partitionList = replicaAssignmentList.split(",")
     val ret = new mutable.HashMap[Int, List[Int]]()
     var partitionId = startPartitionId
-    partitionList = partitionList.takeRight(partitionList.size - partitionId)
-    for (i <- 0 until partitionList.size) {
+    partitionList = partitionList.takeRight(partitionList.length - partitionId)
+    for (i <- partitionList.indices) {
       val brokerList = partitionList(i).split(":").map(s => s.trim().toInt)
-      if (brokerList.size <= 0)
+      if (brokerList.length <= 0)
         throw new AdminOperationException("replication factor must be larger than 0")
-      if (brokerList.size != brokerList.toSet.size)
+      if (brokerList.length != brokerList.toSet.size)
```

Comment:
I'm not sure there's a strong reason to prefer length over size. There are pros and cons.